cmd/process-agent/collector.go

```diff
@@ -222,10 +222,11 @@ func (l *Collector) messagesToResults(start time.Time, name string, messages []m
 			continue
 		}
 
+		agentVersion, _ := version.Agent()
 		extraHeaders := make(http.Header)
 		extraHeaders.Set(headers.TimestampHeader, strconv.Itoa(int(start.Unix())))
 		extraHeaders.Set(headers.HostHeader, l.cfg.HostName)
-		extraHeaders.Set(headers.ProcessVersionHeader, Version)
+		extraHeaders.Set(headers.ProcessVersionHeader, agentVersion.GetNumber())
```

Comment:
Since we're changing all payloads header here, are we sure that `agent.GetNumber()` returns the same value as `Version`?

Reply:
Both flags use the same `get_version` function in the python script. https://github.com/DataDog/datadog-agent/blob/7993fe79b4c88c323be4b8476083e5dbf680413e/tasks/utils.py#L379

The only difference is that in non-release builds it will include the git hash since the last build, although that isn't included in `GetNumber`. https://github.com/DataDog/datadog-agent/blob/e081bed84866686eeab56a98da5ff9b4b9f03ded/pkg/version/version.go#L85